botocore/credentials.py

```diff
@@ -123,6 +126,15 @@ def __init__(self, access_key, secret_key, token=None,
             method = 'explicit'
         self.method = method
 
+    def _assert_ascii(self, s):
+        # Possible Non-ASCII in input, which presumably came from an incorrect
+        # copy-and-paste, would cause confusing UnicodeDecodeError later.
+        # We intercept it here, and provide a meaningful message to customer.
+        try:
+            s.decode('ascii')
+        except UnicodeDecodeError:
+            raise ValueError('Non-ASCII character found in credential: ' + s)
```

Comment:
I wonder if putting the credentials in the exception message has any security implications? If we think most of the errors here will be copy paste, then the provided credentials should be valid credentials + some invalid characters. Are you concerned that we would potentially be dumping out plain-text credentials to logfiles?